pydantic/_internal/_core_metadata.py

```diff
@@ -2,21 +2,23 @@
 
 import inspect
 import typing
+import warnings
 from typing import Any
 
 import typing_extensions
 from pydantic_core import CoreSchema, core_schema
 
-from ._typing_extra import EllipsisType
-
 if typing.TYPE_CHECKING:
-    from ..json_schema import JsonSchemaMetadata, JsonSchemaValue
+    from ..json_schema import JsonSchemaValue
 
 
 # CS is shorthand for Core Schema
 # JS is shorthand for JSON Schema
-_UPDATE_CS_FUNCTION_FIELD = 'pydantic_update_core_schema_function'
-_JS_METADATA_FIELD = 'pydantic_json_schema_metadata'
+_CS_UPDATE_FUNCTION_FIELD = 'pydantic_cs_update_function'
```

Comment:
is there no way for this to be a dataclass or typeddict?

Reply:
a comment on each value/attribute and its meaning would be useful here.

Reply:
I'll convert it to a TypedDict for now